libminifi/test/rocksdb-tests/DBContentRepositoryTests.cpp

```diff
@@ -159,7 +159,7 @@ TEST_CASE("Delete NonExistent Claim", "[TestDBCR4]") {
 
   stream->close();
 
-  content_repo->stop();
+  content_repo->deinitialize();
```

Comment:
Shouldn't this just be called the destructor? What's the reason for keeping it separate?

Reply:
the resource claim stores a shared_ptr to the content repository, "stop()" previously put the repo into an "empty" state, so that a new repository instance could open and manage the rocksdb backend, (during normal operations a ResourceClaim should not outlive the content repo), but these tests depend on being able to invalidate a previous instance and reusing an old claim, these tests should be revised

Reply:
And why do we need a separate stop() and deinitialize(), which closes the DB after stop. Why can't stop() close the DB like before?

Reply:
for the database content repository `stop` was a test-only feature, since it had no background threads there wasn't
really a point in following in flowfilerepository's footsteps
now that it has a background thread its lifecycle should mimic the flowfilerepository/provenancerepository's  `ctor -> initialize -> start <-> stop -> dtor` pattern IMO

Reply:
we could drop the deinitialize (or make it private) and allow the tests to access it